envoy/upstream/upstream.h

```diff
@@ -220,6 +220,12 @@ class Host : virtual public HostDescription {
    */
   virtual void setOutlierDetector(Outlier::DetectorHostMonitorPtr&& outlier_detector) PURE;
 
+  /**
+   * Set the timestamp of when the host has transitioned from unhealthy to healthy state via an
+   * active healchecking.
+   */
+  virtual void setLastHcPassTime(MonotonicTime&& last_hc_pass_time) PURE;
```

Comment:
nit: realistically it will likely be faster to pass this by value as it's just an integer